app/utils/document-delete-helpers.js

```diff
@@ -22,6 +22,10 @@ export async function deleteDocumentContainer(documentContainerOrPromise) {
         await deletePiece(latestPiece);
       } else if (latestPiece.constructor.modelName === 'report') {
         await deleteReport(latestPiece);
+      } else {
+        console.debug(
+          'Piece subclass might not be removed propery, add override to document-delete-helpers.js'
+        );
```

Comment:
do we need console debug? we only need to add minutes modelname later and we are good.

Reply:
I would argue you can still forget if you don't use this helper.
If you do use this helper for future sub-classes, you should look at what the code does and if it's suitable for your situation.
In that case, A comment would have been enough.

We could also add comments to the resource file for piece but this is not exclusive to piece. Perhaps we will have more polymorphism later on with a similar issue.
Perhaps we could extend the readme file and add some notes on polymorphism in the frontend.

In hindsight, we could probably avoid using the modelName and just make deletes optional.
we try to load `piece.pieceParts`. If it's empty, no delete should happen.

Reply:
What is the problem with having a log message printed anyway?

Reply:
not really a problem if debug flag is not on. fine as-is